### PR TITLE
WIP: Replace hand-rolled CLI usage parser with a library

### PR DESCRIPTION
**Pull request by PsychoLlama**
Having very few dependencies is nice. Maintaining a hand-rolled parser
is not. I'm migrating away from my hand-written usage parser onto
a parser combinator library called [parsimmon](https://github.com/jneen/parsimmon).

### CODE REVIEW

**Comment by PsychoLlama** [comment]
Ugh. I tried several different parsing libraries and all of them have frustrating downsides. Whatever library I choose damages the developer experience. My biggest gripe is parser error messages. Every library is extremely limited in what you can customize.

These are the libraries I evaluated:
- [nearley](https://github.com/kach/nearley) (BNF)
  I'm not a huge fan of the EJS-style JS node processor embeddings or the explosion in build step complexity. And the resulting parser is "just okay". As I mentioned before, error messages are barely configurable. The dispute usage grammar is simple and doesn't demand much performance, so the relatively extreme tradeoffs aren't worth it.
- [parsimmon](https://github.com/jneen/parsimmon) (an old parser combinator)
  I've got experience with Parsimmon. It's a fun little library. But once again, the error messages aren't very configurable. You're limited by the power of `.chain(...)` and `.fail(...)` OR `.desc(...)` which is pretty useless when you're dealing with token enums or several simultaneous error conditions.
- [parjs](https://github.com/GregRos/parjs) (a newer parser combinator)
  This is the nicest library available. It has **fantastic** TypeScript support and encourages good patterns with parser combinators. It's very powerful, but it has the same limitations as parsimmon. Then there's the issue of stability ([see this, for example](https://github.com/GregRos/parjs/issues/22#issuecomment-559573923)). It's a very young library.

So in conclusion, yeah I could do it, but the code improvement isn't really that notable, and the end user's experience might actually suffer because of it. It isn't worth the regression or the added dependency.

So I'm giving up 😞
